src/pages/Locations/LocationDetails/LocationDetails.jsx

```diff
@@ -0,0 +1,40 @@
+import {useEffect, useState} from 'react';
+import {useLocation} from "react-router-dom";
+import useLocationPhotos from "../../../hook/useLocationPhotos";
+import './LocationDetails.css';
+import LocationTitle from "./components/LocationTitle/LocationTitle";
+import LocationPhotosGrid from "./components/LocationPhotosGrid/LocationPhotosGrid";
+
+
+const LocationDetails = () => {
+    const {state} = useLocation();
+    const {location, locationName, locationId} = state;
+    const locationRegion = location.region ? location.region + ", " : ''
+    const locationCountry = location.country ?? ''
+    const {requestCallback, locationPhotos} = useLocationPhotos()
+    useEffect(() => {
+        if (locationPhotos == null) {
+            requestCallback(locationId);
+        }
+    }, [locationId, locationPhotos, requestCallback])
+    const [photos, setPhotos] = useState(null)
+    let x = 0
+    useEffect(() => {
+        if (locationPhotos != null) {
+            setPhotos([])
+            while (x < locationPhotos.length) {
+                let y = locationPhotos[x].prefix + "1080x900" + locationPhotos[x].suffix
+                setPhotos(arr => [...arr, y])
+                x++
```

Comment:
Codacy has a fix for the issue: [Replace `················x++` with `········x++;`](https://app.codacy.com/gh/mohammadnedaei/Nearby-Locations-React/pullRequest?prid=9855911)
```suggestion
        x++;
```